core/authentication/fake.go

```diff
@@ -31,24 +30,30 @@ func NewFake() *FakeAuth {
 
 // NewContext adds authentication details to a new background context.
 func (a *FakeAuth) NewContext(userID string) context.Context {
-	md := make(map[string][]string)
-	md["userid"] = []string{userID}
-	return metadata.NewOutgoingContext(context.TODO(), md)
+	return InsertFakeAuthIntoContext(context.TODO(), userID)
+}
+
+// Includes (fake) authentication information in the context to authenticate to the server as userID.
+func InsertFakeAuthIntoContext(ctx context.Context, userID string) context.Context {
```

Comment:
Try making `FakeAuth` implement the `credentials.PerRPCCredentials` interface. 
That way, it could work seamlessly with the other GRPC authentication libraries.